### ISSUE TITLE
TabView: Forward tab button is clipped in 200% text scaling mode.

### ISSUE THREAD

**Issue by YuliKl** [microsoft/microsoft-ui-xaml#3527]
**Describe the bug**

**Steps to reproduce the bug**

Pre-requisite: 
- Settings--> Ease of Access-->Display-->Drag the Display slider  to  200%-->Apply
- Screen Resolution Should be "1280*768"

1. Launch XAML Control Gallery app.
2. Navigate to TabView button and activate it.
3. Navigate to TabStripHeader and TabStripFooter content region->add some more tabs so forward and backward tab button appears-> navigate to forward/backward tab button observe in 200% text scaling.

**Expected behavior**

forward/backward tab button should be visible in 200% text scaling mode .

**Screenshots**

![image](https://user-images.githubusercontent.com/13142665/97658112-bef9fb80-1a28-11eb-8ad6-83f66fa2fb35.png)

**Version Info**

Xaml Controls Gallery version 1.2.16.0 

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_queries/edit/23645026)

**Comment by Felix-Dev** [comment]
Also see issue https://github.com/microsoft/microsoft-ui-xaml/issues/3144. I plan to take a look at that next week. If I don't find the time, I'll let the team know so somebody else can look into it.

**Comment by kmgallahan** [comment]
Ref: #1650 #1519 #1413 and [this comment](https://github.com/microsoft/microsoft-ui-xaml/issues/1650#issuecomment-557232921).

**Comment by Felix-Dev** [comment]
Adding IsTextScaleFactorEnabled=False to the TabView scroll buttons gives us the following result with 200% text scaling mode:
![image](https://user-images.githubusercontent.com/1398851/97778281-9253fa00-1b76-11eb-8306-2bc1ece6c46a.png)

We currently align the scroll buttons to the bottom of the tab strip and control the width/height of the scroll button via theme resources (TabViewItemScrollButtonHeight/TabViewItemScrollButtonWidth). If the resulting placement of the scroll buttons in relation to the other TabView UI elements is not desired, we could change the vertical alignment of the scroll button to center:
![image](https://user-images.githubusercontent.com/1398851/97778400-2faf2e00-1b77-11eb-8f6b-4ddfde1c25ff.png)

This is not yet perfect either so we could add an additional top margin of 10 for better alignment with the rest of the TabView elements (tabs, Add Tab button). The margin of 10 was chosen so that in 100% text scaling, the scroll button has the same position as currently:

| Text scaling | 100% | 200% |
|--|--|--|
||![image](https://user-images.githubusercontent.com/1398851/97798075-1499f800-1c23-11eb-9274-9fa25245aebe.png)|![image](https://user-images.githubusercontent.com/1398851/97798097-3eebb580-1c23-11eb-97e1-253e1f3e56e9.png)|

@YuliKl What do you think? It could be a bit weird looking to customers that changing the text scaling will also change the position of the scroll buttons (in this case, move the button "up" from the tab strip bottom).

(If we choose to use this custom margin should we expose it via a theme resource? I.e. TabViewItemScrollButtonMargin)

On that note, outside of text scaling we also notice the current bottom alignment of the scroll buttons when customers change the height of the TabView items. For example, if a customer changes the height to 60, they will be greeted by this UI: 
![image](https://user-images.githubusercontent.com/1398851/97798353-ac004a80-1c25-11eb-855f-530de9cdffee.png)

Using the possible center vertical alignment with a top margin of 10 for the scroll buttons, the UI looks like this:
![image](https://user-images.githubusercontent.com/1398851/97798368-c2a6a180-1c25-11eb-8ebb-720fe6a22825.png)

I don't know about other customers, but to me the centered scroll buttons (with top margin) look better than the bottom aligned scroll buttons.

**Comment by chingucoding** [comment]
What about having all buttons be aligned to the bottom:

![image](https://user-images.githubusercontent.com/16122379/97800802-892b6180-1c38-11eb-94ad-697909e7eaf9.png)

Having a button centered but not take up the vertical space seems a bit odd to me.

**Comment by Felix-Dev** [comment]
Looking at how I would naturally tab tabs (touch screen), I would tab these tabs at the height of their header. So in case of the scroll buttons (and even the Add button) I would then have to move my finger down a bit, as opposed to having them centered. (The Add tab button would still work since its height is the same as the height of the tab but I would naturally be inclined to tab the button where its icon is located on the screen). 

So in the case of touch screen, I woudl definitely prefer these buttons to be (visually) aligned with the headers of the tab items so that I can keep my finger at the same height level.

**Comment by chingucoding** [comment]
I guess in the touch screen scenario it is more convenient, however, you would be left with inconsistency in the sizing behavior. If the pointer is not near the buttons, you see this:

![image](https://user-images.githubusercontent.com/16122379/97801291-e4ab1e80-1c3b-11eb-9ddd-0ca26f722f2b.png)

But if users try to hover of them, users will be surprised to see that one icon is only centered, while the other stretches the complete height:
![image](https://user-images.githubusercontent.com/16122379/97801285-d65d0280-1c3b-11eb-9c3b-115d3c736cf4.png)

If those two buttons like almost identical in the normal state, I would expect them to behave the same, not one stretching while the other just only covers a portion of the expected height. Also for mouse users, it's more convenient to have the buttons on the bottom since they are faster to reach with the cursor that way when having done something in the content of the TabView.

**Comment by Felix-Dev** [comment]
> But if users try to hover of them, users will be surprised to see that one icon is only centered, while the other stretches the complete height:

I am assuming the scroll buttons not stretching vertically like the Add button is desired though since we've introduced theme resources to customize the width and height of the scroll buttons. That design could be changed so we have matching button designs for the TabView. Edge UWP for example uses more or less the same design for the scroll and add buttons.

I am writing these ideas under the premise that the current design is the correct one (so "fixed" height for the scroll buttons) and in that case I prefer vertically centering (+ the additional top margin) the scroll buttons for visual alignment. It just looks visually off to me otherwise if the scroll button icons are not on the same baseline like the rest of the TabView UI elements.

Firefox for example does it the other way around. For them, the scroll buttons stretch vertically but the Add Tab button is centered vertically:
![firefox-scale](https://user-images.githubusercontent.com/1398851/97801927-5ab18480-1c40-11eb-88dd-1daf637c29af.gif)

So, I would either prefer verticall centering or changing the design of the scroll buttons to match that of the Add Tab button. Though that could conflict with the already introduced theme resources and might not be the chosen design.

**Comment by chingucoding** [comment]
I think the introduction of those themeresources was done without accounting for this kind of scenario. Given that those resources where introduced quite some time now, we will need to figure a way out of fixing this without removing said resources or ignoring them. Since text scale changing is nothing I would expect developers to listen too to account for the correct sizing, the only feasable options I see would be:
1. Align all buttons to the bottom
2. Center "Add tab" button and have it not stretch entire height.

**Comment by Felix-Dev** [comment]
I will use this occasion to bring these design discussions to the attention of @mdtauk since this could be interesting to him as well.

**Comment by mdtauk** [comment]
> 
> Firefox for example does it the other way around. For them, the scroll buttons stretch vertically but the Add Tab button is centered vertically:
> ![firefox-scale](https://user-images.githubusercontent.com/1398851/97801927-5ab18480-1c40-11eb-88dd-1daf637c29af.gif)
> 


Your mention of the way Firefox did it, strikes me as a good solution.  I would ask how this might affect Windows Terminal, which uses a split drop-down button for their add tab UI. How would other content alignment with the tabs and scroll buttons?

**Comment by Felix-Dev** [comment]
@mdtauk Terminal titlebar currently looks like this with 200% text scaling:
![terminal-200-text-scaling](https://user-images.githubusercontent.com/1398851/97885542-c91d4200-1d27-11eb-8006-f23dd735787b.gif)

So a vertically stretching scroll button would fit nicely with the rest of the UI.

**Comment by chingucoding** [comment]
The problem is that having the scroll buttons stretch will be almost impossible to achieve without ignore the existing theme resources/API for that :/

**Comment by chigy** [comment]
We will be tracking this in our [internal issue](https://microsoft.visualstudio.com/OS/_workitems/edit/33595713)

**Comment by mrlacey** [comment]
> We will be tracking this in our [internal issue](https://microsoft.visualstudio.com/OS/_workitems/edit/33595713)

But why does that mean this should be closed?

**Comment by chigy** [comment]
Sure

**Comment by karenbtlai** [comment]
Closing - does not repro with updated font icons ![image](https://user-images.githubusercontent.com/7976322/134205507-0c19dc26-289a-4b3d-84ca-c8aba5f90173.png)
